kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -268,6 +268,23 @@ impl<'tcx> GotocCtx<'tcx> {
             }};
         }
 
+        /// Gets the basename of an intrinsic given its trimmed name.
+        ///
+        /// For example, given `arith_offset::<u8>` this returns `arith_offset`.
+        fn intrinsic_basename(name: &str) -> &str {
+            let scope_sep_count = name.matches("::").count();
+            // We expect at most one `::` separator from trimmed intrinsic names
+            assert!(
```

Comment:
Can you please change this to `debug_assert`? In this case, delay the error to be a verification one iff the intrinsic is actually reachable.

Reply:
Done.